src/main/kotlin/frc/robot/subsystems/shooter/Shooter.kt

```diff
@@ -4,10 +4,37 @@ import edu.wpi.first.units.*
 import edu.wpi.first.wpilibj.Timer
 import edu.wpi.first.wpilibj2.command.Command
 import edu.wpi.first.wpilibj2.command.SubsystemBase
+import frc.robot.lib.LoggedTunableNumber
 import org.littletonrobotics.junction.AutoLogOutput
 import org.littletonrobotics.junction.Logger
 
 class Shooter private constructor(private val io: ShooterIO) : SubsystemBase() {
+    private val topKP: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Top kP", ShooterConstants.TOP_GAINS.kP)
+    private val topKI: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Top kI", ShooterConstants.TOP_GAINS.kI)
+    private val topKD: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Top kD", ShooterConstants.TOP_GAINS.kD)
+    private val topKS: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Top kS", ShooterConstants.TOP_GAINS.kS)
+    private val topKV: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Top kV", ShooterConstants.TOP_GAINS.kV)
+    private val topKA: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Top kA", ShooterConstants.TOP_GAINS.kA)
+
+    private val bottomKP: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Bottom kP", ShooterConstants.BOTTOM_GAINS.kP)
+    private val bottomKI: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Bottom kI", ShooterConstants.BOTTOM_GAINS.kI)
+    private val bottomKD: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Bottom kD", ShooterConstants.BOTTOM_GAINS.kD)
+    private val bottomKS: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Bottom kS", ShooterConstants.BOTTOM_GAINS.kS)
+    private val bottomKV: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Bottom kV", ShooterConstants.BOTTOM_GAINS.kV)
+    private val bottomKA: LoggedTunableNumber =
+        LoggedTunableNumber("Shooter/Bottom kA", ShooterConstants.BOTTOM_GAINS.kA)
```

Comment:
I believe you can get rid of the type declaration

Reply:
99cb3adfa8e0aaf29d1b256ccf275bbe06b7b8fd